galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java

```diff
@@ -457,7 +493,90 @@ public IZosBatchJob getRegionJob() throws CicstsManagerException {
 
     @Override
     public void alterSit(@NotNull String sitParam, String sitValue) throws CicstsManagerException {
-        throw new CicstsManagerException("This method is yet to be developed");
+
+    	// Get the CICS
+    	ICicsRegion theCics = cicstsManager.locateCicsRegion(this.getTag());
+    	
+    	// Restricted SIT parameter
+    	if (sitParam.toUpperCase().equals("APPLID") | sitParam.toUpperCase().equals("GRPLIST") || sitParam.toUpperCase().equals("SYSID")) {
+    		throw new CicstsManagerException(sitParam.toUpperCase() + " is a restricted SIT parameter. It cannot be changed");
+    	}
+    	
+    	// CICS is currently running
+        if (job.getStatus() == JobStatus.ACTIVE) {
+        	throw new CicstsManagerException("The CICS region " + theCics.getApplid() + " with tag " + theCics.getTag() + " is still running. Cannot alter a running CICS region");
+        }
+        
+        for (CICSRegion cicsRegion : complex.getCICS()) {
+        	
+        	// Find the correct region
+        	if (cicsRegion.getApplid().equals(theCics.getApplid())) {
+        		
+        		// Add the Sit parameter
+        		if (sitValue != null) {
+        			
+					Sit sitOverride = new Sit(sitParam, sitValue, "Overridden during runtime");
+
+					// Apply the SIT override
+					cicsRegion.runtimeOverrideSIT(sitOverride);
+	        		rebuildRuntimeJob(cicsRegion);
+        		
+        		} else {
+        			
+        			throw new CicstsManagerException("Sit value in alterSit() cannot be null use removeSit()");
+        		}
+           	}
+        }
     }
+    
+    @Override
+    public void removeSit(@NotNull String sitParam) throws CicstsManagerException {
+    	
+    	// Get the CICS
+    	ICicsRegion theCics = cicstsManager.locateCicsRegion(this.getTag());
+    	
+    	// Check the restricted parameters
+    	if (sitParam.toUpperCase().equals("APPLID") | sitParam.toUpperCase().equals("GRPLIST") || sitParam.toUpperCase().equals("SYSID")) {
+    		throw new CicstsManagerException(sitParam.toUpperCase() + " is a restricted SIT parameter. It cannot be removed");
+       	}
```

Comment:
There is a concern here that if we wanted to test what happened if you tried to start a CICS without these parameters we might have a problem, but I think we are OK